java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducer.java

```diff
@@ -86,7 +86,25 @@
     private final KinesisProducerConfiguration config;
     private final Map<String, String> env;
     private final AtomicLong messageNumber = new AtomicLong(1);
-    private final Map<Long, SettableFuture<?>> futures = new ConcurrentHashMap<>();
+    private final Map<Long, SettableFutureTracking> futures = new ConcurrentHashMap<>();
+
+    private class SettableFutureTracking {
```

Comment:
I adhered the current implementation which is not static for the future map to avoid affecting applications who might be using multiple KinesisProducer instances. I would not want to change this to static

Reply:
Oh i didn't mean to change the `futures` map. Make SettableFutureTracking as private static class, as it doesn't need access to enclosing class.

Reply:
Done